modules/custom/inope_policy_review/inope_policy_review.module

```diff
@@ -144,36 +234,284 @@ function inope_policy_review_form($form, &$form_state, $node) {
-    // TODO: put this into the database.
-  }
-  // If 'I have reviewed this policy and would like to see changes' is selected.
-  elseif ($form_state['values']['inope_policy_review_status_radio'] == 1) {
+  // PART 2 - Check to see if we are updating an existing entry by checking if
+  // the form value hidden_policy_review_id is set to a non empty string.
+  if ($form_state['values']['inope_policy_review_hidden_policy_review_id'] != '') {
+
+    // Run the update query using form_state values.
+    inope_policy_review_update_query($form_state);
+
+    // Clear previous messages and thank the user for their update.
+    drupal_get_messages();
     drupal_set_message(
-      t('User : @userid : @username submitted the following change request:', array(
-        '@userid' => $account->uid,
+      t('Thank you @username for your update to the policy', array(
         '@username' => $account->name,
       )));
 
-    drupal_set_message($form_state['values']['inope_policy_review_changes_requested_textarea']);
+    // Send the user back to the main review page for the node.
+    drupal_goto('node/' . $node->nid . '/review');
+
+  }
+  else {
+    // PART 3 - If 'I have reviewed this policy and approved it for use' is
+    // selected then insert values into the database.
+    if ($form_state['values']['inope_policy_review_status_radio'] == 0) {
+
+      $policy_review_id = inope_policy_review_insert_query($node, $account, '', 0);
+
+      if ($policy_review_id != 0) {
+
+        // Clear previous messages and thank the user for their approval.
+        drupal_get_messages();
+        drupal_set_message(
+          t('Thank you @username for approving the policy', array(
+            '@username' => $account->name,
+          )));
+      }
+    }
+    // PART 4 - If 'I have reviewed this policy and would like to see changes'
+    // is selected then insert values into the database.
+    elseif ($form_state['values']['inope_policy_review_status_radio'] == 1) {
+
+      $policy_review_id = inope_policy_review_insert_query($node, $account, $text, 1);
+
+      if ($policy_review_id != 0) {
+
+        // Clear previous messages and thank the user for their request.
+        drupal_get_messages();
+        drupal_set_message(
+          t('Thank you @username for submitting the following change request:', array(
+            '@username' => $account->name,
+          )));
+
+        // Check_plain happened when this variable was declared.
+        drupal_set_message($text);
+      }
+    }
+    // PART 5 - Catch errors.
+    else {
+      drupal_set_message(t('Form state not recognized, please contact your
+        administrator'));
+    }
+  }
+}
+
+/**
+ * Select query for the current node revision timestamp.
+ */
+function inope_policy_review_select_query_timestamp($node) {
+
+  // Select query to get the latest node revision timestamp.
+  $node_revision = db_query("
+  SELECT vid, timestamp FROM {node_revision} WHERE vid = :vid", array(
```

Comment:
If you truly want the latest timestamp for a given vid -- and you believe you may have multiple entries for a given VID that's probably a MAX -- given our modules, i don't think we'll run into that.